### PR TITLE
Cleanup internal/testcomponents

### PR DESCRIPTION
**Pull request by bogdandrutu**
* Expose example processor to allow start/shutdown tests.

Signed-off-by: Bogdan Drutu <bogdandrutu@gmail.com>

### CODE REVIEW

File: internal/testcomponents/example_exporter.go [hunk 1]
```diff
@@ -25,81 +25,76 @@ import (
...
 	component.WithTracesExporter(createTracesExporter),
 	component.WithMetricsExporter(createMetricsExporter),
 	component.WithLogsExporter(createLogsExporter))
 
-// CreateDefaultConfig creates the default configuration for the Exporter.
 func createExporterDefaultConfig() config.Exporter {
-	return &ExampleExporter{
+	return &struct {
+		config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
+	}{
```
Comment (tigrannajaryan) [inline internal/testcomponents/example_exporter.go:41]: This looks different from what we do with real components. Perhaps for the example to be closer to the reality keep the ExampleExporterConfig struct declared at the top lexical level.

File: internal/testcomponents/example_exporter.go [hunk 2]
```diff
@@ -25,81 +25,76 @@ import (
...
-// ExampleExporter is for testing purposes. We are defining an example config and factory
-// for "exampleexporter" exporter type.
-type ExampleExporter struct {
-	config.ExporterSettings `mapstructure:",squash"` // squash ensures fields are correctly decoded in embedded struct
-}
-
 const expType = "exampleexporter"
 
-// ExampleExporterFactory is factory for ExampleExporter.
+// ExampleExporterFactory is factory for ExampleExporterConfig.
```
[the remark this answers is not in the document]
Comment (bogdandrutu) [reply internal/testcomponents/example_exporter.go]: Mistake, thanks for catching this.

File: internal/testcomponents/example_exporter.go (hunk 1 above)
Comment (bogdandrutu) [reply internal/testcomponents/example_exporter.go]: This is not an example despite the name, it is used for testing. But will update to follow the standard.

File: internal/testcomponents/example_processor_test.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	rcv := &ExampleProcessor{}
+	host := componenttest.NewNopHost()
+	assert.False(t, rcv.Started)
+	err := rcv.Start(context.Background(), host)
+	assert.NoError(t, err)
+	assert.True(t, rcv.Started)
+
+	err = rcv.Shutdown(context.Background())
+	assert.NoError(t, err)
+	assert.True(t, rcv.Started)
```
Comment (fatsheep9146) [inline internal/testcomponents/example_processor_test.go:36]: I think these lines should be same as other tests

```
	assert.False(t, rcv.Stopped)
	err = rcv.Shutdown(context.Background())
	assert.NoError(t, err)
	assert.True(t, rcv.Stopped)
```

Comment (bogdandrutu) [reply internal/testcomponents/example_processor_test.go]: Thanks, done.

**Review by mx-psi — APPROVED** [review]

**Pull request merged**